src/vm/jitinterface.cpp

```diff
@@ -11623,13 +11623,21 @@ void CEEJitInfo::allocMem (
 
     S_SIZE_T totalSize = S_SIZE_T(codeSize);
 
+    bool roData16bytesAligned = ((flag & CORJIT_ALLOCMEM_FLG_RODATA_16BYTE_ALIGN)!= 0);
     if (roDataSize > 0)
     {
-        totalSize.AlignUp(sizeof(void *));
+        if (roData16bytesAligned)
+        {
+            totalSize.AlignUp(16);
+        }
+        else
+        {
+            totalSize.AlignUp(sizeof(void *));
+        }
         totalSize += roDataSize;
 
 #ifndef _WIN64
-        if (roDataSize >= 8)
+        if ((roDataSize >= 8) && !roData16bytesAligned)
```

Comment:
I don't understand. Before my change the code under #ifndef _WIN64 was adding 4 bytes to totalSize if the roData alignment was 8 bytes. That was necessary because right before #ifndef _WIN64 total size was aligned up to pointer size (i.e., 4 in this case). With the new flag roData alignment is 16 bytes and totalSize is aligned up to 16 bytes so no extra padding is necessary. What am I missing?